tests/test_flash_messages.py

```diff
@@ -0,0 +1,187 @@
+    """
+    post_message_url = f"{BASE_URL}/add_message"
+    data = {"text": message_text}
+    return user_session.post(post_message_url, data=data, allow_redirects=True)
+
+def logout_user(user_session=None):
+    """
+    Log out the current user.
+
+    :param user_session: Optional requests session
+    :return: Response object from the request
+    """
+    logout_url = f"{BASE_URL}/logout"
+    return user_session.get(logout_url, allow_redirects=True)
+
+def clean_database():
+    """
+    Truncate all tables in the database to clean the state.
+    """
+    with psycopg2.connect(DATABASE_URL) as conn:
+        with conn.cursor() as cur:
+            cur.execute("TRUNCATE TABLE users CASCADE;")
+            cur.execute("TRUNCATE TABLE followers CASCADE;")
+            cur.execute("TRUNCATE TABLE messages CASCADE;")
+            conn.commit()
+
+@pytest.fixture(scope="module")
+def user1_session():
+    """
+    Create and return a session for user1. This session persists for all tests in the module.
+    """
+    session = requests.Session()
+    return session
+
+@pytest.fixture(scope="module", autouse=True)
+def fetch_public_page():
+    """
+    Fetches the public page before all tests as health check
+    """
+    public_page_url = f"{BASE_URL}/public"
+    response = requests.get(public_page_url)
+    assert response.status_code == 200
+    return response
+
+@pytest.fixture(scope="session", autouse=True)
+def cleanup_db_after_tests():
+    """
+    Automatically clean the database after all tests in the session.
+    """
+    yield  # Let all tests execute
+    clean_database()
+
+def test_register_flash(user1_session):
+    """
+    Test flash message for user registration.
+    """
+    expected_flash = "You were successfully registered and can login now"
+    response = register_user("user1", "user1@waect.com", "waect", user1_session)
+    assert response.status_code == 200, "register failed"
+    assert expected_flash in response.text
+
+def test_login_flash(user1_session):
+    """
+    Test flash message for user login.
+    """
+    expected_flash = "You were logged in"
+    response = login_user("user1", "waect", user1_session)
+    assert response.status_code == 200, "login failed"
+    assert expected_flash in response.text
+
+def test_user1_follow_user2_flash(user1_session):
+    """
+    Test flash message for user1 following user2.
+    """
+    user2_session = requests.Session()
+    register_user("user2", "user2@waect.com", "waect", user2_session)
+    expected_flash = f'You are now following &#34;user2&#34;'
+    response = follow_user("user2", user1_session)
+    assert response.status_code == 200, "follow failed"
+    assert expected_flash in response.text
+
+def test_user1_follow_user2_already_followed_flash(user1_session):
+    """
+    Test flash message for user1 following user2 when already followed.
+    """
+    expected_flash = 'You are already following &#34;user2&#34;'
```

Comment:
This is not a flash-message in Helge's service.

https://github.com/search?q=repo%3AHelgeCPH%2Fcpython_energy+flash&type=code

Reply:
1. 
Helge uses:
```
flash(f'You are no longer following "{username}"')
```
Also we defined that in our wiki: https://github.com/WAEC-T/wiki/blob/main/02_Areas/RequirementsSpecification/Flash-messages.md

2. 
Good catch will delete that! What are you guys throwing there?